CppSamples/Layers/ConfigureElectronicNavigationalCharts/ConfigureElectronicNavigationalCharts.cpp

```diff
@@ -0,0 +1,346 @@
+}
+
+ConfigureElectronicNavigationalCharts::~ConfigureElectronicNavigationalCharts()
+{
+  // Reset settings to avoid affecting other samples.
+  EncDisplaySettings* globalDisplaySettings = EncEnvironmentSettings::displaySettings();
+  globalDisplaySettings->marinerSettings()->resetToDefaults();
+  globalDisplaySettings->viewingGroupSettings()->resetToDefaults();
+  globalDisplaySettings->textGroupVisibilitySettings()->resetToDefaults();
+}
+
+void ConfigureElectronicNavigationalCharts::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<ConfigureElectronicNavigationalCharts>("Esri.Samples", 1, 0, "ConfigureElectronicNavigationalChartsSample");
+}
+
+MapQuickView* ConfigureElectronicNavigationalCharts::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void ConfigureElectronicNavigationalCharts::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+  {
+    return;
+  }
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  // Initialize callout data (following ShowCallout pattern)
+  m_mapView->calloutData()->setVisible(false);
+  m_mapView->calloutData()->setTitle("ENC Feature");
+
+  // Connect to mouse clicked events for feature identification
+  connect(m_mapView, &MapQuickView::mouseClicked, this, &ConfigureElectronicNavigationalCharts::onGeoViewTapped);
+
+  // Connect to doneLoading signal of EncExchangeSet
+  connect(m_encExchangeSet, &EncExchangeSet::doneLoading, this, [this](const Error& error)
+  {
+    if (!error.isEmpty())
+    {
+      return;
+    }
+
+    // Loop through datasets
+    const QList<EncDataset*> datasets = m_encExchangeSet->datasets();
+    for (EncDataset* dataset : datasets)
+    {
+      // Create an EncCell from each dataset
+      EncCell* cell = new EncCell(dataset, this);
+
+      // Create an EncLayer from each cell
+      EncLayer* encLayer = new EncLayer(cell, this);
+
+      // Connect to the doneLoading signal of the layer
+      connect(encLayer, &EncLayer::doneLoading, this, [this, encLayer](const Error& error)
+      {
+        if (!error.isEmpty())
+        {
+          return;
+        }
+
+        m_extents << encLayer->fullExtent();
+
+        if (m_extents.length() != m_encExchangeSet->datasets().length())
+        {
+          return;
+        }
+
+        // Combine the extents
+        Envelope fullExtent = GeometryEngine::combineExtents(m_extents);
+        m_mapView->setViewpointAsync(Viewpoint{fullExtent});
+      });
+
+      // Add the layer to the map
+      m_map->operationalLayers()->append(encLayer);
+    }
+  });
+
+  // Load the EncExchangeSet
+  m_encExchangeSet->load();
+
+  emit mapViewChanged();
+}
+
+EncColorScheme ConfigureElectronicNavigationalCharts::colorScheme() const
+{
+  return m_colorScheme;
+}
+
+void ConfigureElectronicNavigationalCharts::setColorScheme(Esri::ArcGISRuntime::EncColorScheme colorScheme)
```

Comment:
```suggestion
void ConfigureElectronicNavigationalCharts::setColorScheme(EncColorScheme colorScheme)
```

Here and throughout. We are using the namespace.